tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+'''
+Utility module to read and write Prometheus Alert rules to file.
+Ensures we can test the Alert rules deployed and for documentation
+purposes.
+'''
+
+import argparse
+import functools
+import ipaddress
+import json
+import operator
+import requests
+import subprocess
+import sys
+import yaml
+
+
+class AlertRuleFile:
```

Comment:
I don't really see the point of having a class here and it's poorly named, the class is mainly a wrapper around requests on Prometheus API.